binheap.py

```diff
@@ -17,11 +17,10 @@ def push(self, value):
 
     def __parent(self, index):
```

Comment:
Do you intend to have Python name-mangling invoked here?  It's a bit odd to do, but acceptable.  I just want to verify that you know what you're doing when you prefix an attribute name with double underscores.